src/components/views/dialogs/MessageEditHistoryDialog.js

```diff
@@ -45,10 +46,17 @@ export default class MessageEditHistoryDialog extends React.PureComponent {
         const opts = {from: this.state.nextBatch};
         const roomId = this.props.mxEvent.getRoomId();
         const eventId = this.props.mxEvent.getId();
-        const result = await MatrixClientPeg.get().relations(
-            roomId, eventId, "m.replace", "m.room.message", opts);
+        let result;
         let resolve;
-        const promise = new Promise(r => resolve = r);
+        let reject;
+        const promise = new Promise((_resolve, _reject) => {resolve = _resolve; reject = _reject;});
```

Comment:
Hmm, I wish `setState` could be awaited, as it would make a lot of this cleaner... Anyway, just a random thought.

Reply:
Indeed. Fwiw, this method needs to return a promise because scrollpanel calls it when paginating (and doesn't call it again until the promise resolves or rejects), so that's why we need to wait for the state to be updated.